go/multipooler/manager/rpc_consensus.go

```diff
@@ -55,20 +60,19 @@ func (pm *MultiPoolerManager) BeginTerm(ctx context.Context, req *consensusdatap
 	}
 
 	// Check if we've already accepted a different candidate in this term
-	if req.Term == currentTerm && acceptedLeader != nil && acceptedLeader.GetName() != req.CandidateId {
+	if req.Term == currentTerm && acceptedLeader != "" && acceptedLeader != req.CandidateId {
 		return response, nil
 	}
 
-	// Validate and update term if needed (handles req.Term >= currentTerm)
-	// This will update the term to req.Term if req.Term > currentTerm, resetting AcceptedLeader to nil
-	// If req.Term == currentTerm, it does nothing (which is correct for idempotent acceptance)
-	if err := pm.validateAndUpdateTerm(ctx, req.Term, false); err != nil {
-		return nil, fmt.Errorf("failed to validate term: %w", err)
+	// Update term if needed (only if req.Term > currentTerm)
+	// This will reset AcceptedLeader to nil
+	if req.Term > currentTerm {
```

Comment:
We can remove all this block right? Because now we are doing the Prepare and Save below

Reply:
We could combine them as an optimization. I tried to keep them separate so that Revocation and Acceptance (aka voting) are kept logically separate.
We discussed this offline, but there's still a gap here, because we need to also revoke at the postgres level by clearing out the primaryConnInfo on a standby (but only if this pooler is a standby). That is still TODO and planned for a future PR.